packages/expo-av/android/build.gradle

```diff
@@ -90,6 +131,30 @@ dependencies {
 
   implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:${getKotlinVersion()}"
 
+  //noinspection GradleDynamicVersion
+  implementation 'com.facebook.react:react-native:+'
+
+  compileOnly 'com.facebook.fbjni:fbjni:0.2.2'
+  extractHeaders 'com.facebook.fbjni:fbjni:0.2.2:headers'
+  extractJNI 'com.facebook.fbjni:fbjni:0.2.2'
+  def rnAARs = fileTree("${REACT_NATIVE_DIR}/android").matching { include "**/*.aar" }
+  if (rnAARs.any()) {
+    // node_modules/react-native has a .aar, extract headers
+    if (rnAARs.size() > 1) {
+      logger.error("More than one React Native AAR file has been found:")
+      rnAARs.each {println(it) }
+      throw new GradleException("Multiple React Native AARs found:\n${rnAARs.join("\n")}" +
+          "\nRemove the old ones and try again")
+    }
+    def rnAAR = rnAARs.singleFile
+    extractJNI(files(rnAAR))
+  } else {
+    // there is no prebuilt react-native.aar, we must be building from source. directly include the headers
+    def rnAndroid = fileTree("${REACT_NATIVE_DIR}/ReactAndroid")
+    // TODO: Build from ReactAndroid/ and React/ headers source?
+    logger.error("node_modules/react-native/android does not contain a *.aar file! Cannot build expo-av. TODO: Build from ReactAndroid/ source?")
```

Comment:
for expo go, we don't have the aar. expo go build still works here because the `REACT_NATIVE_DIR` is getting through the `require.resolve('react-native/package.json')` and it's not the `react-native-lab/react-native` right now. the might be a risk if we have different react-native versions between bare-expo and expo-go. for expo-go, we should have different logic to extract the react-native so files.

FYI: for expo go to build the vendored reanimated, we get the react-native so files from `ReactAndroid/build`:
https://github.com/expo/expo/blob/f395b7f877c07de8ba44459ef7f61d6e85075d5a/android/expoview/build.gradle#L65

Reply:
So if I understand correctly, you mean that we should delete the `else` clause here (so ignore if there are no AARs) and in CMake check both locations for `.so` files: the existing one from extracted AARs and if it doesn't exist, then the one you suggested (`RN_SO_DIR`)?

Reply:
Cool, it seems to work this way, just needed to add `${ANDROID_ABI}` to the AAR path on the CMake side. Thanks! 😁